### DISCUSSION TITLE
Style Editor with ChatGPT

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre#243] [Cross-project feature proposals]
<img width="50px" src="https://user-images.githubusercontent.com/53421382/228279318-7162615f-502e-48ac-befb-45ae4e00102d.png"/>

Playing around a bit with ChatGPT, I noticed that it knows the style specification quite well already. For example, you can tell it to generate a style json showing houses in black at zoom 15 and higher and it will do it.

My proposal would be to create a full style editor using ChatGPT or similar with live preview and custom data sources.

It would be a bit like maputnik, just without buttons...

**Comment by HarelM** [comment]
I think it's a great idea, you'd still need to inspect the tile data in order to have chatGPT give you better answers, and also I've seen it fail miserably on some cases, but a preview would solve this issues I believe.

**Reply by wipfli** [reply]
Ideally you could supply a tilejson to ChatGPT or some sort of schema definition file as an input.

**Reply by wipfli** [reply]
Preview would help, maybe also show the console to the user and feed console errors automatically back to the large language model.

**Reply by wipfli** [reply]
Basically create a loop where the renderer, the language model, and the user interact.

**Comment by HarelM** [comment]
This can get a bounty under the Lab direction maybe.
